front/lib/api/signup.ts

```diff
@@ -195,16 +170,16 @@ export async function handleEnterpriseSignUpFlow(
     }),
   ]);
 
-  // Early return if user is already a member of a workspace.
-  if (total !== 0) {
-    return { flow: null, workspace: null };
-  }
-
   // Redirect to login error flow if workspace is not found.
   if (!workspace) {
     return { flow: "unauthorized", workspace: null };
   }
```

Comment:
the workspace must exist and user must be part of it, otherwise enterprise login is not authorized